### PR TITLE
Fix `ColorPicker` shape icon is invisible until shape is changed

### LINKED ISSUE #84519: ColorPicker shape icon is invisible until shape is changed
### Godot version

4.2 beta4

### System information

Windows 10.0.19045 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1060 (NVIDIA; 30.0.15.1403) - Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 Threads)

### Issue description

Can't see icon:
image
After change:
image

### Steps to reproduce

1. Add any node property of type Color (e.g. Node2D)
2. Open color picker

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by DinDotDout**
Fixes #84519

### CODE REVIEW

File: scene/gui/color_picker.cpp [hunk 1]
```diff
@@ -89,6 +89,8 @@ void ColorPicker::_notification(int p_what) {
 			shape_popup->set_item_icon(shape_popup->get_item_index(SHAPE_VHS_CIRCLE), theme_cache.shape_circle);
 			shape_popup->set_item_icon(shape_popup->get_item_index(SHAPE_OKHSL_CIRCLE), theme_cache.shape_circle);
 
+			btn_shape->set_icon(shape_popup->get_item_icon(current_shape));
```
Comment (YuriSizov) [inline scene/gui/color_picker.cpp:92]:
```suggestion
			if (current_shape != SHAPE_NONE) {
				btn_shape->set_icon(shape_popup->get_item_icon(current_shape));
			}
```

There is this case when the shape is hidden completely, then this won't work. So we need to guard against it.

Also, could you please amend the commit message. The current title of this PR is a good option that follows our preferred style. When you apply this suggestion, make sure to squash/fixup your commits into one.

**Review by YuriSizov — APPROVED** [review]
Looks good to me!

Unrelated to this PR since this is what existing code looks like, but it seems like we're abusing the fact that IDs and indices of the shape popup match. Sometimes we're using `SHAPE_*` constants to get the index and sometimes we're using them directly as index.

This should be cleaned up at some point (but again, that's not something this PR introduces or needs to address).

**Comment by DinDotDout** [comment]
Ah i see they are being added into a vector in an order matching their enum value. I still need to familiarize myself with C++, setup my editor properly and test a faster building system but i could look into it when i am a bit more versed in a new pr.

**Comment by KoBeWi** [comment]
There are plans for refactoring the picker shapes entirely, as the remaining part of https://github.com/godotengine/godot-proposals/issues/4353, but that's a bigger task.

**Comment by DinDotDout** [comment]
I noticed that hsl is the only mode that still displays a choser when none is set. I have not dug into the code but they may be hardcoded together

**Comment by YuriSizov** [comment]
Thanks! And congrats on your first merged Godot PR!

**Pull request merged**